aws/resource_aws_backup_vault.go

```diff
@@ -0,0 +1,120 @@
+package aws
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/backup"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsBackupVault() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsBackupVaultCreate,
+		Read:   resourceAwsBackupVaultRead,
+		Update: resourceAwsBackupVaultUpdate,
+		Delete: resourceAwsBackupVaultDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`[a-z0-9\-]+`), "must contain alphanumeric characters or underscores"),
+			},
+			"tags": {
```

Comment:
For proper drift detection, this attribute should have `d.Set("tags", ...)` called during `resourceAwsBackupVaultRead`, likely using [`ListTags`](https://docs.aws.amazon.com/sdk-for-go/api/service/backup/#Backup.ListTags)